ingest/src/main/java/com/microsoft/azure/kusto/ingest/IngestionProperties.java

```diff
@@ -301,40 +308,50 @@ void validate() throws IngestionClientException {
 
         String mappingReference = ingestionMapping.getIngestionMappingReference();
         IngestionMapping.IngestionMappingKind ingestionMappingKind = ingestionMapping.getIngestionMappingKind();
-        String message = null;
+        String message = "";
 
-        if (ingestionMapping.getColumnMappings() != null) {
-            if (StringUtils.isNotBlank(mappingReference)) {
-                message += String.format("Both mapping reference '%s' and column mappings were defined.", mappingReference);
-            }
-            for (ColumnMapping column : ingestionMapping.getColumnMappings()) {
-                if (!column.isValid(ingestionMappingKind)) {
-                    message += String.format("Column mapping '%s' is invalid", column.columnName);
-                }
-            }
-        } else if (StringUtils.isBlank(mappingReference)) {
+        if ((ingestionMapping.getColumnMappings() == null) && StringUtils.isBlank(mappingReference)) {
             if (dataFormat.isMappingRequired()) {
                 message += String.format("Mapping must be specified for '%s' format.", dataFormat.name());
```

Comment:
Maybe there should be a newline between concatenated messages?

Reply:
Good idea